app/services/catalog_search_service.rb

```diff
@@ -0,0 +1,42 @@
+# Uses the Blacklight JSON API to search and then extracts select Catalog fields
+class CatalogSearchService < AbstractSearchService
+  def initialize(options = {})
+    options[:query_url] ||= Settings.CATALOG_QUERY_URL.to_s
+    options[:response_class] ||= Response
+    super
+  end
+
+  class Response < AbstractSearchService::Response
+    def total
+      @json ||= JSON.parse(@body)
+      @json['response']['pages']['total_count'].to_i
+    end
+
+    def results
+      @json ||= JSON.parse(@body)
+      solr_docs = @json['response']['docs']
+      solr_docs.collect do |doc|
+        result = OpenStruct.new
```

Comment:
What would you think about using a PORO or a Struct here?  

http://blog.honeybadger.io/how-openstruct-and-hashes-can-kill-performance/